src/iperf_sctp.c

```diff
@@ -486,16 +492,17 @@ iperf_sctp_connect(struct iperf_test *test)
         i_errno = IESTREAMCONNECT;
         return -1;
     }
-    freeaddrinfo(server_res);
 
     /* Send cookie for verification */
     if (Nwrite(s, test->cookie, COOKIE_SIZE, Psctp) < 0) {
 	saved_errno = errno;
 	close(s);
+	freeaddrinfo(server_res);
 	errno = saved_errno;
         i_errno = IESENDCOOKIE;
         return -1;
     }
+    freeaddrinfo(server_res);
```

Comment:
This addition here will cause a double-free if the `setsockopt()` call a few lines down fails...its error-handling code also does `freeaddrinfo(server_res)`.  Perhaps this new line should move below that block (but before the `return s`).

Reply:
Right, totally missed that.